gladier/client.py

```diff
@@ -202,92 +145,28 @@ def scopes(self):
-        if getattr(self, '_flows_client', None) is not None:
-            return self._flows_client
-
-        automate_authorizer = self.authorizers[
-            globus_automate_client.flows_client.MANAGE_FLOWS_SCOPE
-        ]
-        flow_scope = self.get_section(private=True).get('flow_scope')
-        flow_authorizer = self.authorizers.get(flow_scope)
-
-        def get_flow_authorizer(*args, **kwargs):
-            return flow_authorizer
-
-        self._flows_client = globus_automate_client.FlowsClient.new_client(
-            self.client_id, get_flow_authorizer, automate_authorizer,
-        )
-        return self._flows_client
-
-    @property
-    def funcx_client(self):
-        """
-        :return: an authorized funcx client
-        """
-        if getattr(self, '__funcx_client', None) is not None:
-            return self.__funcx_client
-
-        login_manager = gladier.utils.funcx_login_manager.FuncXLoginManager(authorizers=self.authorizers)
-        self.__funcx_client = FuncXClient(login_manager=login_manager)
-        return self.__funcx_client
-
-    def login(self, **login_kwargs):
-        """Login to the Gladier client. This will ensure the user has the correct
-        tokens configured but it DOES NOT guarantee they are in the correct group to
-        run a flow. Can be run both locally and on a server.
-        See help(fair_research_login.NativeClient.login) for a full list of kwargs.
-        """
-        nc = self.get_native_client()
-        if self.is_logged_in() and login_kwargs.get('force') is not True:
-            log.debug('Already logged in, skipping login.')
-            return
-        log.info('Initiating Native App Login...')
-        log.debug(f'Requesting Scopes: {self.scopes}')
-        login_kwargs['requested_scopes'] = login_kwargs.get('requested_scopes', self.scopes)
-        login_kwargs['refresh_tokens'] = login_kwargs.get('refresh_tokens', True)
-        nc.login(**login_kwargs)
-        self.authorizers = nc.get_authorizers_by_scope()
-        self._flows_client = None
+        return self.login_manager.missing_authorizers
 
     def logout(self):
         """Log out and revoke this client's tokens. This object will no longer
         be usable until a new login is called.
         """
-        if not self.client_id:
-            raise gladier.exc.AuthException('Gladier client must be instantiated with a '
-                                            '"client_id" to use "login()!')
-        log.info(f'Revoking the following scopes: {self.scopes}')
-        self.get_native_client().logout()
-        # Clear authorizers cache
-        self.authorizers = dict()
+        return self.login_manager.logout()
 
     def is_logged_in(self):
         """
         Check if the client is logged in.
 
         :return: True, if there are no self.missing_authorizers. False otherwise.
         """
-        return not bool(self.missing_authorizers)
+        return self.login_manager.is_logged_in()
```

Comment:
These are left in for backwards compatibility, but we may want to nix them in the future.